daml-lf/spec/daml-lf-1.rst

```diff
@@ -2493,48 +2375,258 @@ This section lists the built-in functions supported by DAML LF 1.
+      𝕆('LESS_EQ' @σ b₁ b₂) = Ok (¬b₁ ∨ b₂)
+
+    —————————————————————————————————————————————————————————————————————— EvLessEqInt64
+      𝕆('LESS_EQ' @σ LitInt64₁ LitInt64₂) = Ok (LitInt64₁ ≤ₗ LitInt64₂)
+
+    —————————————————————————————————————————————————————————————————————— EvLessEqDate
+      𝕆('LESS_EQ' @σ LitDate₁ LitDate₂) = Ok (LitDate₁ ≤ₗ LitDate₂)
+
+    —————————————————————————————————————————————————————————————————————— EvLessEqTimestamp
+      𝕆('LESS_EQ' @σ LitTimestamp₁ LitTimestamp₂) =
+          Ok (LitTimestamp₁ ≤ LitTimestamp₂)
+
+    —————————————————————————————————————————————————————————————————————— EvLessEqText
+      𝕆('LESS_EQ' @σ LitText₁ LitText₂) = Ok (LitText₁ ≤ₗ LitText₂)
+
+    —————————————————————————————————————————————————————————————————————— EvLessEqParty
+      𝕆('LESS_EQ' @σ LitParty₁ LitParty₂) = Ok (LitParty₁ ≤ₗ LitParty₂)
+
+    —————————————————————————————————————————————————————————————————————— EvLessEqNumeric
+      𝕆('LESS_EQ' @σ LitNumeric₁ LitNumeric₂) =
+          Ok (LitNumeric₁ ≤ LitNumeric₂)
+
+    —————————————————————————————————————————————————————————————————————— EvLessEqStructEmpty
+      𝕆('LESS_EQ' @⟨ ⟩ ⟨ ⟩ ⟨ ⟩) = Ok 'True'
+
+      𝕆('LESS_EQ' @τ₁ v₀ v₀') = Err t
+    —————————————————————————————————————————————————————————————————————— EvLessEqStructNonEmptyHeadErr1
+      𝕆('LESS_EQ' @⟨ f₀: τ₁,  f₁: τ₁, …,  fₙ: τₙ ⟩
+                   ⟨ f₀= v₀,  f₁= v₁, …,  fₘ= vₘ ⟩
+   	           ⟨ f₀= v₀', f₁= v₁', …, fₘ= vₘ' ⟩) = Err t
+
+      𝕆('LESS_EQ' @τ₁ v₀ v₀') = Ok 'False'
+    ————————————————————————————————————————————————————————————————————— EvLessEqStructNonEmptyHeadBigger
+      𝕆('LESS_EQ' @⟨ f₀: τ₁,  f₁: τ₁, …,  fₙ: τₙ  ⟩
+                   ⟨ f₀= v₀,  f₁= v₁, …,  fₘ= vₘ  ⟩
+      	           ⟨ f₀= v₀', f₁= v₁', …, fₘ= vₘ' ⟩) = Ok 'False'
+
+      𝕆('LESS_EQ' @τ₁ v₀ v₀') = Ok 'True'
+      𝕆('LESS_EQ' @τ₁ v₀' v₀) = Err t
+    —————————————————————————————————————————————————————————————————————— EvLessEqStructNonEmptyHeadErr2
+      𝕆('LESS_EQ' @⟨ f₀: τ₁,  f₁: τ₁, …,  fₙ: τₙ  ⟩
+                   ⟨ f₀= v₀,  f₁= v₁, …,  fₘ= vₘ  ⟩
+      	           ⟨ f₀= v₀', f₁= v₁', …, fₘ= vₘ' ⟩) = Err t
+
+      𝕆('LESS_EQ' @τ₁ v₀ v₀') = Ok 'True'
+      𝕆('LESS_EQ' @τ₁ v₀' v₀) = Ok 'False'
+    ————————————————————————————————————————————————————————————————————— EvLessEqStructNonEmptyHeadSmaller
+      𝕆('LESS_EQ' @⟨ f₀: τ₁,  f₁: τ₁, …,  fₙ: τₙ  ⟩
+                   ⟨ f₀= v₀,  f₁= v₁, …,  fₘ= vₘ  ⟩
+   	           ⟨ f₀= v₀', f₁= v₁', …, fₘ= vₘ' ⟩) = Ok 'True'
+
+      𝕆('LESS_EQ' @τ₁ v₀ v₀') = Ok 'True'
+      𝕆('LESS_EQ' @τ₁ v₀' v₀) = Ok 'True'
+      𝕆('LESS_EQ' @⟨ f₁: τ₁, …,  fₙ: τₙ  ⟩
+                   ⟨ f₁= v₁, …,  fₘ= vₘ  ⟩
+   	           ⟨ f₁= v₁', …, fₘ= vₘ' ⟩) = r
+    —————————————————————————————————————————————————————————————————————— EvLessEqStructNonEmptyTail
+      'LESS_EQ' @⟨ f₀: τ₁,  f₁: τ₁, …,  fₙ: τₙ ⟩
```

Comment:
```suggestion
      'LESS_EQ' @⟨ f₀: τ₀,  f₁: τ₁, …,  fₙ: τₙ ⟩
```

Reply:
And the same change to all the previous struct head rules.

Reply:
Good catch. Thanks.